indexing-hadoop/src/main/java/org/apache/druid/indexer/HadoopIndexTask.java

```diff
@@ -833,7 +831,7 @@ public String runTask(String[] args) throws Exception
       // can be injected based on the configuration given in config.getSchema().getIOConfig().getMetadataUpdateSpec()
       final SegmentMetadataPublisher maybeHandler;
       if (config.isUpdaterJobSpecSet()) {
-        maybeHandler = new SegmentMetadataPublisher(INJECTOR.getInstance(IndexerMetadataStorageCoordinator.class));
+        maybeHandler = new SegmentMetadataPublisher(HadoopTask.INJECTOR.getInstance(IndexerMetadataStorageCoordinator.class));
```

Comment:
Just saw that `HadoopTask` has its own static `INJECTOR` and `EXTENSION_LOADER`. 😅 
Didn't go through the whole thing but seems like a really hacky approach to load a bunch of classes.

Reply:
yea, this isn't new to this PR 😅. i suppose this line change isn't really needed and was just a side-effect of moving stuff around, but will leave it unless i need to push more commits to this branch